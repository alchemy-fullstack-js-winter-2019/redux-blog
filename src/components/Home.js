import React from 'react';
import PropTypes from 'prop-types';

export default function Home({ posts }) {
  const postList = posts.map(post => <li key={post.id}>{post.title}</li>);
  return <ul>{postList}</ul>;
}

Home.propTypes = {
  posts: PropTypes.array.isRequired
};
