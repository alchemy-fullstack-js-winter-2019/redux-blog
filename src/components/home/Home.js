import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getPostsAction } from '../../actions/posts';
import { getLongestPost } from '../../selectors/posts';
import { Link } from 'react-router-dom';

class Home extends PureComponent {
  static propTypes = {
    posts: PropTypes.array.isRequired,
    fetchPosts: PropTypes.func.isRequired
  }

  componentDidMount() {
    this.props.fetchPosts();
  }
  
  render() {
    const { posts } = this.props;
    const postList = posts.map((post, i) => {
      return <Link key={i} to={`/posts/${post.userId}/${post.id}`}><li key={i}>{post.body}</li></Link>;
    });
    return (
      <>
      <h1> HOME COMPONENT </h1>
      <ul>{postList}</ul>
      </>
    );
  }
}

const mapStateToProps = state => ({
  posts: getLongestPost(state)
});

const mapDispatchToProps = dispatch => ({
  fetchPosts() {
    dispatch(getPostsAction());
  }
});


export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);
