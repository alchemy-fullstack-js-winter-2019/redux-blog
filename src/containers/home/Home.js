import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getLongestPosts } from '../../selectors/posts';
import LongestPosts from '../../components/posts/LongestPosts';
import { fetchPosts } from '../../actions/posts';

class Home extends PureComponent {
  static propTypes = {
    posts: PropTypes.array.isRequired,
    fetch: PropTypes.func.isRequired
  };

  componentDidMount() { 
    this.props.fetch();
  }

  render() {
    return (
      <LongestPosts {...this.props}/>
    );
  }
}

const mapStateToProps = (state) => ({
  posts: getLongestPosts(state)
});

const mapDispatchToProps = dispatch => ({
  fetch() {
    dispatch(fetchPosts());
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);
