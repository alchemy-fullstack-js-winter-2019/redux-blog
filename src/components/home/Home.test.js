import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Home component tests', () => {
  it('renders the Home component correctly', () => {
    const posts = [{ body: 'something', id: 1 }, { body: 'something', id: 2 }];
    const tree = renderer.create(
      <Router>
        <Home posts={posts} />
      </Router>
    );
    expect(tree).toMatchSnapshot;
  });
});
