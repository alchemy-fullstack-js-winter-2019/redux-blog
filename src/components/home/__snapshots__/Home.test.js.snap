// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches a snapshot 1`] = `
Array [
  <h2>
    FEATURED POSTS
  </h2>,
  <ul>
    <li>
      <a
        className="link"
        href="/posts/undefined"
        onClick={[Function]}
      >
        <h3>
          title
        </h3>
      </a>
      <p
        className="postBody"
      >
        body
      </p>
    </li>
  </ul>,
]
`;
